src/P3_processes.jl

```diff
@@ -99,3 +99,131 @@ function ice_melt(
+
+Provides a function `ρ′_rim(Dᵢ, Dₗ)` that computes the local rime density [kg/m³] 
+    for a given ice particle diameter `Dᵢ` [m] and liquid particle diameter `Dₗ` [m].
+
+# Arguments
+- `state`: [`P3State`](@ref)
+- `velocity_params`: velocity parameterization, e.g. [`CMP.Chen2022VelType`](@ref)
+- `ρₐ`: air density [kg/m³]
+- `T`: temperature [K]
+
+# Returns
+A function that computes the local rime density [kg/m³] using the equation:
+
+```math
+ρ'_{rim} = 51 + 114 R_i - 5.5 R_i^2
+```
+where
+```math
+R_i = \\frac{ 10^6 ⋅ D_{liq} ⋅ |v_{liq} - v_{ice}| }{ 2 T_{sfc} }
+```
+and ``T_{sfc}`` is the surface temperature [°C], ``D_{liq}`` is the liquid particle
+diameter [m], ``v_{liq/ice}`` is the particle terminal velocity [m/s].
+So the units of ``R_i`` are [m² s⁻¹ °C⁻¹]. The units of ``ρ'_{rim}`` are [kg/m³].
+
+They assume for simplicity that ``T_{sfc}`` equals ``T``, the ambient air temperature.
+For real graupel, ``T_{sfc}`` is slightly higher than ``T`` due to latent heat release 
+of freezing liquid particles onto the ice particle. Morrison & Milbrandt (2013) 
+found little sensitivity to "realistic" increases in ``T_{sfc}``.
+
+Implementation follows Cober and List (1993), Eq. 16 and 17.
+See also the P3 fortran code, `microphy_p3.f90`, Line 3315-3323,
+which extends the range of the calculation to ``R_i ≤ 12``, the upper limit of which
+then equals the solid bulk ice density, ``ρ^⭒ = 900 kg/m^3``.
+
+Note that Morrison & Milbrandt (2015) [MorrisonMilbrandt2015](@cite) only uses this 
+parameterization for collisions with cloud droplets.
+For rain drops, they use the solid bulk ice density, ``ρ^* = 900 kg/m^3``.
+We do not consider this distinction, and use this parameterization for all liquid particles.
+"""
+function compute_local_rime_density(state, velocity_params, ρₐ, T)
+    (; T_freeze) = state.params
+    T°C = T - T_freeze  # Convert to °C
+    μm = 1_000_000  # Note: m to μm factor, c.f. units of rₘ in Eq. 16 in Cober and List (1993)
+
+    v_ice = ice_particle_terminal_velocity(state, velocity_params, ρₐ)
+    v_liq = CO.particle_terminal_velocity(velocity_params.rain, ρₐ)
+    function ρ′_rim(Dᵢ, Dₗ)
+        v_term = abs(v_ice(Dᵢ) - v_liq(Dₗ))
+        Rᵢ = (Dₗ * μm * v_term) / (2 * T°C)  # Eq. 16 in Cober and List (1993). Note: no `-` due to absolute value in v_term
+        return ρ′_rim_P3(Rᵢ)
+    end
+    return ρ′_rim
+end
+
+"""
+    ρ′_rim_P3(Rᵢ)
+
+Returns the local rime density [kg/m³] for a given Rᵢ [m² s⁻¹ °C⁻¹].
+
+Based on Cober and List (1993), Eq. 16 and 17 (valid `for 1 ≤ Rᵢ ≤ 8`).
+For `8 < Rᵢ ≤ 12`, linearly interpolate between `ρ′_rim(8) ≡ 611 kg/m³` and `ρ⭒ = 900 kg/m³`.
+See also the P3 fortran code, `microphy_p3.f90`, Line 3315-3323.
+"""
+function ρ′_rim_P3(Rᵢ)
+    # TODO: Externalize these parameters
+    a, b, c = 51, 114, -11 // 2 # coeffs for Eq. 17 in Cober and List (1993), converted to [kg / m³]
+    ρ⭒ = 900  # ρ^⭒: density of solid bulk ice
```

Comment:
We should probably be using the ice density we already have in [parameters](https://github.com/CliMA/ClimaParams.jl/blob/c66047dcc360f288780d76ca3a40dfbb4aad3a12/src/parameters.toml#L2134) ?